src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,194 @@
+        nodeBefore.prev.next = newNode;
+        nodeBefore.prev = newNode;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> nodeToModify = getNodeByIndex(index);
+        T returnValue = nodeToModify.item;
+        nodeToModify.item = value;
+        return returnValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T returnValue = unlink(getNodeByIndex(index));
+        size--;
+        return returnValue;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> nodeToRemove = getNodeByValue(object);
+        if (nodeToRemove == null) {
+            return false;
+        }
+        unlink(nodeToRemove);
+        size--;
+        return true;
+    }
+
+    private T unlink(Node<T> nodeToRemove) {
+        T returnValue;
+        if (nodeToRemove == first) {
+            returnValue = removeFirst();
+        } else if (nodeToRemove == last) {
+            returnValue = removeLast();
+        } else {
+            returnValue = nodeToRemove.item;
+            nodeToRemove.prev.next = nodeToRemove.next;
+            nodeToRemove.next.prev = nodeToRemove.prev;
+        }
+        return returnValue;
+    }
+
+    private T removeFirst() {
+        T returnValue = first.item;
+        if (size == 1) {
+            first = null;
+        } else if (size == 2) {
+            first = last;
+            last = null;
+        } else {
+            first = first.next;
+            first.prev = null;
+        }
+        return returnValue;
+    }
+
+    private T removeLast() {
+        T returnValue = last.item;
+        if (size == 2) {
+            first.next = null;
+            last = null;
+        } else {
+            last = last.prev;
+            last.next = null;
+        }
+        return returnValue;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Invalid index " + index);
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> pointer;
+        if (index <= size / 2) {
+            pointer = first;
+            for (int i = 0; i < index; i++) {
+                pointer = pointer.next;
+            }
+        } else {
+            pointer = last;
+            for (int i = size - 1; i > index; i--) {
+                pointer = pointer.prev;
+            }
+        }
+        return pointer;
+    }
+
+    private Node<T> getNodeByValue(T value) {
```

Comment:
logic from this method we can move to the remove(T object) and then we can remove this method